eng/pipelines/templates/jobs/smoke-tests.yml

```diff
@@ -0,0 +1,160 @@
+parameters:
+  - name: Nightly
+    type: boolean
+    default: true
+  - name: ServiceDirectory
+    type: string
+    default: ""
+  - name: Artifacts
+    type: object
+    default: []
+
+jobs:
+  - ${{ if eq(parameters.Nightly, false) }}:
+    - job: smoke_test_eligibility
+      displayName: Check Smoke Test Eligibility
+      steps:
+        - ${{ if ne(variables['Skip.Release'], 'true') }}:
+          - pwsh: |
+              ./Initialize-SmokeTests.ps1 -DryRun -CI -Verbose -ServiceDirectory ${{ parameters.ServiceDirectory }}
+              $packages = Get-Content ./run-manifest.json | ConvertFrom-Json
+              $packages | ForEach-Object {
+                Write-Host "Smoke tests will run for $_.Name"
+                Write-Host "##vso[task.setvariable variable=RunSmokeTests;]true"
+              }
+            name: check_smoke_tests_${{ parameters.ServiceDirectory }}
+            displayName: Check smoke test eligibility for ${{ parameters.ServiceDirectory }}
+            workingDirectory: ./common/smoke-test
+
+        - pwsh: |
+            Write-Host "Setting RunSmokeTests to $($env:RunSmokeTests)"
+            Write-Host "##vso[task.setvariable variable=RunSmokeTests;isOutput=true;]$($env:RunSmokeTests)"
```

Comment:
Why not set the variable as output in the previous step?

Reply:
Originally, it was because if Skip.Release is true (or other conditionals I had been using), then the job won't exist, and the subsequent jobs that take a dependency on it will fail the pipeline validation. If I declare it separately, then it always exists for dependsOn.

Now that things are a little simpler, I can just wrap both the job and the dependsOn downstream in the same conditional. Will update.